packages/esm-ward-app/src/ward-workspace/cancel-admission-request-workspace/cancel-admission-request.workspace.tsx

```diff
@@ -0,0 +1,177 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { ResponsiveWrapper, showSnackbar, useAppContext, useSession } from '@openmrs/esm-framework';
+import classNames from 'classnames';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+import useWardLocation from '../../hooks/useWardLocation';
+import type { ObsPayload, WardPatientWorkspaceProps, WardViewContext } from '../../types';
+import { useCreateEncounter } from '../../ward.resource';
+import styles from './cancel-admission-request.scss';
+import WardPatientWorkspaceBanner from '../patient-banner/patient-banner.component';
+
+export default function CancelAdmissionRequestWorkspace({
+  closeWorkspaceWithSavedChanges,
+  wardPatient,
+  promptBeforeClosing,
+}: WardPatientWorkspaceProps) {
+  const { patient } = wardPatient ?? {};
+  const { t } = useTranslation();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const { createEncounter, emrConfiguration, isLoadingEmrConfiguration, errorFetchingEmrConfiguration } =
+    useCreateEncounter();
+  const { currentProvider } = useSession();
+  const { location } = useWardLocation();
+  const { wardPatientGroupDetails } = useAppContext<WardViewContext>('ward-view-context') ?? {};
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        note: z
+          .string()
+          .trim()
+          .min(1, {
+            message: t(
+              'notesRequiredForCancellingRequest',
+              'Notes required for cancelling admission or transfer request',
+            ),
+          }),
+      }),
+    [t],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const formDefaultValues: Partial<FormValues> = {
+    note: '',
+  };
+
+  const {
+    formState: { errors, isDirty },
+    control,
+    handleSubmit,
+  } = useForm<FormValues>({ resolver: zodResolver(zodSchema), defaultValues: formDefaultValues });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+    return () => promptBeforeClosing(null);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setIsSubmitting(true);
+      setShowErrorNotifications(false);
+      const obs: Array<ObsPayload> = [
+        {
+          concept: emrConfiguration?.consultFreeTextCommentsConcept?.uuid,
+          value: values.note,
+        },
+        {
+          concept: emrConfiguration?.admissionDecisionConcept?.uuid,
+          value: {
+            uuid: emrConfiguration?.denyAdmissionConcept?.uuid,
+          },
+        },
+      ];
+
+      createEncounter(patient, emrConfiguration?.cancelADTRequestEncounterType, [
+        {
+          concept: emrConfiguration?.dispositionDescriptor?.dispositionSetConcept?.uuid,
```

Comment:
This actually doesn't belong with an disposition set. The obs can just be stored directly on on the encounter.

Reply:
Wait... I think that's what I'm doing? This array is an array of obs for the encounter.

Reply:
nvm... I think I know what you mean now. It should be just `obs` instead.